permutation.h

```diff
@@ -80,13 +80,19 @@ static inline void rng_init(unsigned thread_num)
         rng_buf = (char*)calloc(1, RNG_BUF_SIZE);
         rand_state = (struct random_data*)calloc(1, sizeof(struct random_data));
         assert(rand_state);
-        assert(initstate_r(thread_num, rng_buf, RNG_BUF_SIZE, rand_state) == 0);
+        if (initstate_r(thread_num, rng_buf, RNG_BUF_SIZE, rand_state) != 0) {
+                perror("initstate_r");
+                exit(1);
```

Comment:
The rest of the code does it this way.  I'd prefer to be consistent with that.  I'm also not sure if err() is available everywhere.  The manpage says, "These functions are nonstandard BSD extensions."